zcncore/transactionauth.go

```diff
@@ -182,25 +182,27 @@ func (ta *TransactionWithAuth) ReadPoolUnlock(fee uint64) (
 }
 
 // StakePoolLock used to lock tokens in a stake pool of a blobber.
-func (ta *TransactionWithAuth) StakePoolLock(blobberID string,
-	lock, fee uint64) (err error) {
+func (ta *TransactionWithAuth) StakePoolLock(providerId string, providerType Provider, lock uint64, fee uint64) error {
 
 	type stakePoolRequest struct {
-		BlobberID string `json:"blobber_id"`
+		ProviderType Provider `json:"provider_type,omitempty"`
+		ProviderID   string   `json:"provider_id,omitempty"`
 	}
 
-	var spr stakePoolRequest
-	spr.BlobberID = blobberID
+	spr := stakePoolRequest{
+		ProviderType: providerType,
+		ProviderID:   providerId,
+	}
```

Comment:
We use the same struct here in 0chain also. Can we move this common repo and import from there in both 0chain and gosdk? 

https://github.com/0chain/0chain/blob/f040238e04ea5f51ede84834b605d8d88abf73d3/code/go/0chain.net/smartcontract/storagesc/stakepool.go#L386-L389

Reply:
yeah, check this discussion https://0chain.slack.com/archives/G01535M3YPP/p1666090620403399?thread_ts=1666018203.760739&cid=G01535M3YPP
this should be refactored in different issue